pkg/channel/consolidated/dispatcher/consumer_message_handler.go

```diff
@@ -69,16 +74,46 @@ func (c consumerMessageHandler) Handle(ctx context.Context, consumerMessage *sar
 	ctx, span := tracing.StartTraceFromMessage(c.logger, ctx, message, consumerMessage.Topic)
 	defer span.End()
 
-	_, err := c.dispatcher.DispatchMessageWithRetries(
+	te := kncloudevents.TypeExtractorTransformer("")
+
+	bufferedMessage, err := buffering.CopyMessage(ctx, message, &te)
+	args := eventingchannels.ReportArgs{
+		Ns:        c.channelNs,
+		EventType: string(te),
+	}
+
+	_ = message.Finish(nil)
+
+	dispatchExecutionInfo, dispatchErr := c.dispatcher.DispatchMessageWithRetries(
 		ctx,
-		message,
+		bufferedMessage,
 		nil,
 		c.sub.Subscriber,
 		c.sub.Reply,
 		c.sub.DeadLetter,
 		c.sub.RetryConfig,
 	)
 
+	err = parseDispatchResultAndReportMetrics(dispatchExecutionInfo, c.reporter, args, dispatchErr)
+
 	// NOTE: only return `true` here if DispatchMessage actually delivered the message.
 	return err == nil, err
 }
+
+// Ideally this should be moved to Eventing
+func parseDispatchResultAndReportMetrics(info *eventingchannels.DispatchExecutionInfo, reporter eventingchannels.StatsReporter, reportArgs eventingchannels.ReportArgs, dispatchErr error) error {
```

Comment:
why can't we do that?

Reply:
I am going to create a new PR pointing to this one showing that I need to copy the same code. This is to justify the change only, will go away.

Reply:
Also will fix test coverage once coed is eliminated and see what is left.